lib/services/database.go

```diff
@@ -134,6 +137,70 @@ func setDBName(meta types.Metadata, firstNamePart string, extraNameParts ...stri
 	return meta
 }
 
+// NewDatabaseFromAzureMySQLServer creates a database resource from an AzureMySQL server.
+func NewDatabaseFromAzureMySQLServer(server *armmysql.Server) (types.Database, error) {
+	if server.Properties == nil || server.Properties.FullyQualifiedDomainName == nil {
+		return nil, trace.BadParameter("empty endpoint")
+	}
+	endpoint := *server.Properties.FullyQualifiedDomainName
+	if server.Name == nil {
+		return nil, trace.BadParameter("empty server name")
+	}
+	name := *server.Name
+
+	metadata, err := MetadataFromAzureMySQLServer(server)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var location string
+	if server.Location != nil {
+		location = *server.Location
+	}
+	return types.NewDatabaseV3(
+		setDBName(types.Metadata{
+			Description: fmt.Sprintf("Azure MySQL server in %v", location),
+			Labels:      labelsFromAzureMySQLServer(server, metadata),
+		}, name),
+		types.DatabaseSpecV3{
+			Protocol: defaults.ProtocolMySQL,
+			URI:      fmt.Sprintf("%v:%v", endpoint, AzureMySQLPort),
+			Azure:    *metadata,
+		})
+}
+
+// NewDatabaseFromAzurePostgresServer creates a database resource from an AzurePostgres server.
+func NewDatabaseFromAzurePostgresServer(server *armpostgresql.Server) (types.Database, error) {
+	if server.Properties == nil || server.Properties.FullyQualifiedDomainName == nil {
+		return nil, trace.BadParameter("empty endpoint")
+	}
+	endpoint := *server.Properties.FullyQualifiedDomainName
+	if server.Name == nil {
+		return nil, trace.BadParameter("empty server name")
+	}
```

Comment:
this check seems to be duplicated maybe we can make a method for `armpostgresql.Server ` validation.